multiqc/modules/adapterRemoval/adapterRemoval.py

```diff
@@ -0,0 +1,311 @@
+                    'full-length_cp', 'truncated_cp']
+        data_pattern = {'total': 0,
+                        'unaligned': 1,
+                        'aligned': 2,
+                        'discarded_m1': 3,
+                        'singleton_m1': 4,
+                        'retained': 6}
+
+        if self.__read_type == 'paired':
+            data_pattern['discarded_m2'] = 5
+            data_pattern['singleton_m2'] = 6
+            if not self.__collapsed:
+                data_pattern['retained'] = 8
+            else:
+                data_pattern['full-length_cp'] = 8
+                data_pattern['truncated_cp'] = 9
+                data_pattern['retained'] = 10
+
+        for field in required:
+            if field in data_pattern:
+                tmp = trim_data[data_pattern[field]]
+                value = tmp.split(': ')[1]
+                self.result_data[field] = int(value)
+            else:
+                self.result_data[field] = 0
+
+        reads_total = self.result_data['total']
+        aligned_total = self.result_data['aligned']
+        unaligned_total = self.result_data['unaligned']
+        if self.__read_type == 'paired':
+            reads_total = self.result_data['total'] * 2
+            aligned_total = self.result_data['aligned'] * 2
+            unaligned_total = self.result_data['unaligned'] * 2
+
+        self.result_data['aligned_total'] = aligned_total
+        self.result_data['unaligned_total'] = unaligned_total
+        self.result_data['reads_total'] = reads_total
+        self.result_data['discarded_total'] = reads_total - self.result_data['retained']
+
+        self.result_data['retained_reads'] = self.result_data['retained'] - self.result_data['singleton_m1'] - self.result_data['singleton_m2']
+        self.result_data['percent_aligned'] = round((float(self.result_data['aligned']) * 100.0) / float(self.result_data['total']), 2)
+
+    def set_len_dist(self, len_dist_data):
+
+        for line in len_dist_data[1:]:
+            l_data = line.rstrip('\n').split('\t')
+            l_data = list(map(int, l_data))
+
+            # initialize file name
+            if self.s_name not in self.len_dist_plot_data['mate1']:
+                self.len_dist_plot_data['mate1'][self.s_name] = dict()
+                self.len_dist_plot_data['mate2'][self.s_name] = dict()
+                self.len_dist_plot_data['singleton'][self.s_name] = dict()
+                self.len_dist_plot_data['collapsed'][self.s_name] = dict()
+                self.len_dist_plot_data['collapsed_truncated'][self.s_name] = dict()
+                self.len_dist_plot_data['discarged'][self.s_name] = dict()
+                self.len_dist_plot_data['all'][self.s_name] = dict()
+
+            if self.__read_type == 'single':
+                if not self.__collapsed:
+                    self.len_dist_plot_data['mate1'][self.s_name][l_data[0]] = l_data[1]
+                    self.len_dist_plot_data['mate2'][self.s_name][l_data[0]] = 0
```

Comment:
I'm slightly unsure about setting these values as `0`. They're not technically zero reads, they're actually `None` (or no data). This probably doesn't matter in practice, but can cause confusion sometimes and is something I have actively tried to avoid in other modules.

Looking at the source files, I think the way that I would have approached this code is more along the lines of the [Bismark module](https://github.com/ewels/MultiQC/blob/master/multiqc/modules/bismark/bismark.py) - set up a [load of regexes](https://github.com/ewels/MultiQC/blob/master/multiqc/modules/bismark/bismark.py#L20-L39) with associated keys, and then [search the file for each regex](https://github.com/ewels/MultiQC/blob/master/multiqc/modules/bismark/bismark.py#L166-L177) and save whatever values are present in the file. Then any missing values will be just that - missing. This is also probably more robust to changes in file format than this, as it doesn't depend on the position on the file - instead directly using the associated label.

Having missing data instead of zeroes also has the advantage of not needing to do special cases in the plotting, as empty categories will be ignored by the core MultiQC plotting functions (though you would may have to handle some missing-data cases). This is relevant to my comments below about removing unused categories for SE data.

What happens if you use `None` instead of `0` here? Would be worth trying at least I think.